src/pages/Preferences.vue

```diff
@@ -78,7 +78,7 @@ export default Vue.extend({
     async proposePreferences() {
       const { reset } = await this.$store.dispatch(
         'preferences/proposePreferences',
-        this.credentials as ServerState,
+        { ...this.credentials as ServerState },
```

Comment:
I guess we need the spread here (doing nothing) to satisfy TypeScript? Not sure if that's worth a comment…

Reply:
I guess we could destructure credentials and essentially achieve the same thing. I made an update, let me know if this ends up being more clear.. perhaps I'm not following the comment about `doing nothing`.

Reply:
The spread is essentially a no-op (`{ ... x }` is basically the same as `x`, at runtime).  I'll look at your changes and see if we need something else.